source/io/module_javaScriptInvoke.js

```diff
@@ -432,18 +442,33 @@
                     throw new Error('The API being accessed for ' + functionName + ' is not valid anymore.');
                 }
                 completionCallbackStatus.retrievalState = completionCallbackRetrievalEnum.RETRIEVED;
-                return generateCompletionCallback(occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus);
+                return generateCompletionCallback(occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus, isInternalFunction);
             };
+
+            if (isInternalFunction) {
+                api.setLabVIEWError = function (statusCode, errorCode, errorMessage) {
+                    Module.coreHelpers.mergeErrors(statusCode, errorCode, errorMessage, errorStatusPointer, errorCodePointer, errorSourcePointer);
+                };
+            }
             return api;
         };
 
+        publicAPI.javaScriptInvoke.registerInternalFunctions = function (functionsToAdd) {
+            Object.keys(functionsToAdd).forEach(function (name) {
+                if (internalFunctionsMap.has(name)) {
+                    throw new Error('Private function already registered for name:' + name);
+                }
+                internalFunctionsMap.set(name, functionsToAdd[name]);
```

Comment:
should also check that typeof functionsToAdd[name] === 'function'

Reply:
done.